### PR TITLE
WT-11971 Fix doc-update task and artifacts archiving

### PR DESCRIPTION
**Pull request by lukech**
There are 2 changes made in this PR:

1. Fix the "with" statement in `test/evergreen/doc_update.py` by passing it a context manager object. 
2. Move the "wiredtiger.github.com" folder/tree into the "wiredtiger" folder/tree so that it can be included in the artifacts by default (for ease of debugging).

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -449,7 +450,7 @@ functions:
 
             # Synchronize the generated documentation with the current one.
             echo "Synchronizing documentation for branch $branch ..."
-            rsync -avq ../wiredtiger/docs-$branch/ $branch/ --delete
+            rsync -avq ../docs-$branch/ $branch/ --delete
```
Comment (ajmorton) [inline test/evergreen.yml:453]: Why are we setting the verbose `-v` and quiet `-q` flags at the same time?

File: test/evergreen/doc_update.py [hunk 2]
```diff
@@ -29,24 +48,23 @@
...
+    )
 try:
     installation = app.get_repo_installation(owner, repo)
-    print(f"{installation.id=} for {owner}/{repo}")
 except Exception as e:
     sys.exit(e)
 
 # Get an installation access token
 try:
     installation_auth = app.get_access_token(installation.id)
```
Comment (ajmorton) [inline test/evergreen/doc_update.py:61]: This scoping seems a bit weird with the `get_access_token` call now sitting in a try-catch by itself.
I know we discussed this last time but could we merge this block and the block below that calls `git push`?

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml:453]: Good point, the "-v" flag can be saved.

**Review by etienneptl — APPROVED** [review]
LGTM!

File: test/evergreen/doc_update.py (hunk 2 above)
Comment (lukech) [reply test/evergreen/doc_update.py:61]: Fair enough. Will update as suggested.

Comment (lukech) [reply test/evergreen/doc_update.py:61]: See [e504420](https://github.com/wiredtiger/wiredtiger/pull/9854/commits/e504420c0c0808a7c41d10fde2ccac39fc3b97e6)

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml]: See [e504420](https://github.com/wiredtiger/wiredtiger/pull/9854/commits/e504420c0c0808a7c41d10fde2ccac39fc3b97e6)

**Comment by lukech** [comment]
@ajmorton @etienneptl I moved the "wiredtiger.github.com" folder out of the "wiredtiger" folder and archived them into separate tarballs. See [e504420](https://github.com/wiredtiger/wiredtiger/pull/9854/commits/e504420c0c0808a7c41d10fde2ccac39fc3b97e6)

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks for the quick turnaround!

**Review by etienneptl — APPROVED** [review]

**Pull request merged**